lib/Doctrine/ORM/Internal/Hydration/ScalarColumnHydrator.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\ORM\Internal\Hydration;
+
+use Doctrine\ORM\Exception\MultipleSelectorsFoundException;
+
+/**
+ * Hydrator that produces one-dimensional array that is often (not always) interchangeable
+ * with the corresponding object for read-only access.
+ */
+class ScalarColumnHydrator extends AbstractHydrator
+{
+    /**
+     * {@inheritdoc}
+     * @throws \Doctrine\ORM\Exception\MultipleSelectorsFoundException
+     * @throws \Doctrine\DBAL\Driver\Exception
+     */
+    protected function hydrateAllData()
+    {
+        if (count($this->_rsm->fieldMappings) > 1) {
+            throw MultipleSelectorsFoundException::create($this->_rsm->fieldMappings);
+        }
+
+        $result = [];
+
+        $column = array_keys($this->_rsm->fieldMappings)[0];
+
+        while ($row = $this->_stmt->fetchOne()) {
+            $result[] = $row[$column];
```

Comment:
Maybe we could simplify it further with `$result[] = current($row)`? That way you do not need to compute `$column`.

Reply:
updated